iothub_client/samples/pnp/pnp_simple_thermostat/pnp_simple_thermostat.c

```diff
@@ -0,0 +1,441 @@
+            // So here we simply need the root of the JSON itself.
+            desiredObject = rootObject;
+        }
+    }
+
+    return desiredObject;
+}
+
+//
+// UpdateTemperatureAndStatistics updates the temperature and min/max/average values
+//
+static void UpdateTemperatureAndStatistics(double desiredTemp, bool* maxTempUpdated)
+{
+    if (desiredTemp > g_maxTemperature)
+    {
+        g_maxTemperature = desiredTemp;
+        *maxTempUpdated = true;
+    }
+    else if (desiredTemp < g_minTemperature)
+    {
+        g_minTemperature = desiredTemp;
+    }
+
+    g_numTemperatureUpdates++;
+    g_allTemperatures += desiredTemp;
+
+    g_currentTemperature = desiredTemp;
+}
+
+// Format string to report the property maximum temperature since reboot.  This is a "read only" property from the
+// service solution's perspective, which means we don't need to include any sort of status codes.
+static const char g_maxTemperatureSinceRebootFormat[] = "{\"maxTempSinceLastReboot\": %.2f }";
+
+// Format string to indicate the device received an update request for the temperature.  Because this is a "writeable"
+// property from the service solution's perspective, we need to return a status code (HTTP status code style) and version
+// for the solution to correlate the request and its status.
+static const char g_targetTemperatureResponseFormat[] = "{\"targetTemperature\": { \"value\": %.2f, \"ac\":%d, \"av\":%d, \"ad\":\"%s\" }}";
+
+
+//
+// SendTargetTemperatureReport sends a PnP property indicating the device has received the desired targetted temperature
+//
+static void SendTargetTemperatureReport(IOTHUB_DEVICE_CLIENT_HANDLE deviceClient, double desiredTemp, int responseStatus, int version, const char* description)
+{
+    IOTHUB_CLIENT_RESULT iothubClientResult;
+    char targetTemperatureResponseProperty[256];
+
+    if (snprintf(targetTemperatureResponseProperty, sizeof(targetTemperatureResponseProperty), g_targetTemperatureResponseFormat, desiredTemp, responseStatus, version, description) < 0)
+    {
+        LogError("snprintf building targetTemperature response failed");
+    }
+    else if ((iothubClientResult = IoTHubDeviceClient_SendReportedState(deviceClient, (const unsigned char*)targetTemperatureResponseProperty, strlen(targetTemperatureResponseProperty), NULL, NULL)) != IOTHUB_CLIENT_OK)
+    {
+        LogError("Unable to send reported state for target temperature.  Error=%d", iothubClientResult);
+    }
+    else
+    {
+        LogInfo("Sending maxTempSinceReboot property");
+    }
+}
+
+//
+// SendMaxTemperatureSinceReboot reports a PnP property indicating the maximum temperature since the last reboot (simulated here by lifetime of executable)
+//
+static void SendMaxTemperatureSinceReboot(IOTHUB_DEVICE_CLIENT_HANDLE deviceClient)
+{
+    IOTHUB_CLIENT_RESULT iothubClientResult;
+    char maxTemperatureSinceRebootProperty[256];
```

Comment:
nit: same as previous consider static

Reply:
This one I can't change.  `SendMaxTemperatureSinceReboot` can be called both from the dispatcher thread and `main` so there'd be a potential race letting both access the same buffer.